### PR TITLE
Sentry

### PR DESCRIPTION
**Pull request by tanneberger**
## Description
Added Sentry at important point where error may occur the stack-trace and given relevant data to find the origin of the bug will be piped into a log file. Which should be than sit in a docker volume. 

## Related Issue
#19 

## Motivation and Context
Test Driven Development

## How Has This Been Tested?
tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by SpartanerSpaten** [comment]
adding sdk

**Comment by tanneberger** [comment]
This introduces new Environment Variables PATH_LOGFILE, DSN, RELEASE

File: README.md [hunk 1]
```diff
@@ -77,13 +78,6 @@ Required are all modules in the `requirements.txt`.
 
 Your microservice will be supported by the [game-server of cryptic](https://github.com/cryptic-game/server).
 
-### Environment variables
```
Comment (felixevers) [inline README.md:80]: Is there a reason for this deletion?

File: cryptic/__init__.py [hunk 2]
```diff
@@ -80,8 +87,51 @@ def set_mode(self, mode):
...
+class Sentry(logging.Logger):
+    def __init__(self, name):
+        super().__init__(self, name)
+        self.__setup_logger()
+        self.__setup_sentry()
+        self.__using_sentry: bool = False
+        self._name: str = name
+
+    def __setup_logger(self) -> None:
+
```
Comment (felixevers) [inline cryptic/__init__.py]: No whitespace at here, please?

File: cryptic/__init__.py [hunk 3]
```diff
@@ -134,7 +184,7 @@ def setup_database(self) -> None:
                 password=_config["MYSQL_PASSWORD"],
                 hostname=_config["MYSQL_HOSTNAME"],
                 port=port,
-                database=_config["MYSQL_DATABASE"]
+                database=_config["MYSQL_DATABASE"],
```
Comment (felixevers) [inline cryptic/__init__.py:186]: Not necessary...

File: cryptic/__init__.py [hunk 4]
```diff
@@ -157,22 +207,22 @@ def ping(self) -> None:
...
     def __init__(self, name: str, server_address: Tuple[str, int] = None):
+        self._sentry: Sentry = Sentry(name)
         self._user_endpoints: Dict[Tuple, Callable] = {}
         self._ms_endpoints: Dict[Tuple, Callable] = {}
         self._user_endpoint_requirements: Dict[Tuple, scheme.Structure] = {}
         self._name: str = name
         self._awaiting = []
         self._data = {}
-        self._database = DatabaseWrapper()
+        self._database: DatabaseWrapper = DatabaseWrapper()
```
Comment (felixevers) [inline cryptic/__init__.py:224]: This is explicit... -> No typing (optional)

File: cryptic/__init__.py [hunk 5]
```diff
@@ -157,22 +207,22 @@ def ping(self) -> None:
...
-            print("Connection Timeout")
+        except Exception as e:
             self.reload()
 
 
 class MicroService:
     SERVICE_REQUEST_MAX_TIMEOUT = 10
 
     def __init__(self, name: str, server_address: Tuple[str, int] = None):
+        self._sentry: Sentry = Sentry(name)
```
Comment (felixevers) [inline cryptic/__init__.py:217]: This is explicit... -> No typing (optional)

File: cryptic/__init__.py [hunk 6]
```diff
@@ -225,76 +275,67 @@ def __exec(self, frame):
...
-                        # MAYBE add sentry here
-                        self.__send({
-                            "tag": tag,
-                            "user": frame["user"],
-                            "data": {
-                                "error": "unknown_service"
-                            }
-                        })
+                        self._sentry.debug("ms requested: " + str(endpoint) + " Endpoint not found")
+                        self.__send({"tag": tag, "ms": frame["ms"], "data": {"error": "unknown_service"}})
```
Comment (felixevers) [inline cryptic/__init__.py:278]: Expand this statement, please!

File: setup.py [hunk 7]
```diff
@@ -3,13 +3,14 @@
...
+    name="cryptic-game",
+    version="0.3.5",
+    description="Microservice handler for cryptic-game",
+    author="cryptic-game",
+    author_email="faq@cryptic-game.net",
+    long_description=file,
+    long_description_content_type="text/markdown",
+    url="https://github.com/cryptic-game/python3-lib",
+    packages=find_packages(),
+)
```
Comment (felixevers) [inline setup.py:26]: <Tab> here, please!

File: setup.py [hunk 8]
```diff
@@ -3,13 +3,14 @@
...
+setup(
+    name="cryptic-game",
+    version="0.3.5",
+    description="Microservice handler for cryptic-game",
+    author="cryptic-game",
+    author_email="faq@cryptic-game.net",
+    long_description=file,
+    long_description_content_type="text/markdown",
+    url="https://github.com/cryptic-game/python3-lib",
+    packages=find_packages(),
```
Comment (felixevers) [inline setup.py:24]: Unnecessary

**Review by felixevers — CHANGES_REQUESTED** [review]
Thank you!
Please improved the code based on this review within a few days.

File: cryptic/__init__.py (hunk 5 above)
Comment (tanneberger) [reply cryptic/__init__.py:217]: Typing == fuck yeah

File: cryptic/__init__.py (hunk 4 above)
Comment (tanneberger) [reply cryptic/__init__.py:224]: Typing == fuck yeah

File: setup.py (hunk 8 above)
Comment (tanneberger) [reply setup.py:24]: needed for pypi complete config

File: README.md (hunk 1 above)
Comment (tanneberger) [reply README.md:80]: yeah will add docu and this table back in

File: cryptic/__init__.py (hunk 6 above)
Comment (tanneberger) [reply cryptic/__init__.py:278]: What do you mean with "Expand this statement" ... should I log more Information or just code clean up @use-to

File: setup.py (hunk 7 above)
Comment (tanneberger) [reply setup.py:26]: What do you mean ?

File: README.md [hunk 9]
```diff
@@ -10,80 +10,43 @@ Pypi Seite: https://pypi.org/project/cryptic-game/
 $ pip3 install cryptic-game
 ```
```
Comment (felixevers) [inline README.md:12]: Can you link the example.py?
```suggestion
## Quick Start
You can follow the [example.py](https://github.com/cryptic-game/python3-lib/blob/master/example.py) to create your first microservice.
```

File: cryptic/__init__.py (hunk 6 above)
Comment (felixevers) [reply cryptic/__init__.py:278]: You can wrap the dict over multiple lines.

File: setup.py (hunk 7 above)
Comment (felixevers) [reply setup.py:26]: Indent this!

**Review by felixevers — CHANGES_REQUESTED** [review]
Better, but just respect my feedback to this pr.
So i dont want to discuss about unnecessary typings at here...

Some little fixes should be done!

File: cryptic/__init__.py (hunk 6 above)
Comment (tanneberger) [reply cryptic/__init__.py:278]: black wants it this way.

**Comment by tanneberger** [comment]
@use-to I am using the [black linter](https://github.com/psf/black) .... and the style changes you suggest are not conform with this pep8 enforcing tool -> black -l 120 . -> black only starts wrapping the dict over multiple lines when the line surpasses 120 characters. It would unnecessary work too check after every code clean up. Black also places the , at the last element of list and inputs and it is pep8 conform so.

File: README.md (hunk 9 above)
Comment (tanneberger) [reply README.md:12]: done

**Review by felixevers — APPROVED** [review]
Nice! Thank you!

**Pull request merged**